neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/impl/RemoteNeo4jSchemaDefiner.java

```diff
@@ -0,0 +1,119 @@
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+import org.hibernate.tool.hbm2ddl.UniqueConstraintSchemaUpdateStrategy;
+import org.neo4j.graphdb.Label;
+
+/**
+ * Initialize the schema for the Neo4j database:
+ * <ol>
+ * <li>create sequences;</li>
+ * <li>create unique constraints on identifiers, natural ids and unique columns</li>
+ * </ol>
+ * <p>
+ * Note that unique constraints involving multiple columns won't be applied because Neo4j does not support it.
+ * <p>
+ * The creation of unique constraints can be skipped setting the property
+ * {@link Environment#UNIQUE_CONSTRAINT_SCHEMA_UPDATE_STRATEGY} to the value
+ * {@link UniqueConstraintSchemaUpdateStrategy#SKIP}. Because in Neo4j unique constraints don't have a name, setting the
+ * value to {@link UniqueConstraintSchemaUpdateStrategy#RECREATE_QUIETLY} or
+ * {@link UniqueConstraintSchemaUpdateStrategy#DROP_RECREATE_QUIETLY} will have the same effect: keep the existing
+ * constraints and create the missing one.
+ *
+ * @author Davide D'Alto
+ * @author Gunnar Morling
+ */
+public class RemoteNeo4jSchemaDefiner extends Neo4jSchemaDefiner<Statements> {
+
+	private static final Log log = LoggerFactory.getLogger();
+
+	@Override
+	public void initializeSchema(SchemaDefinitionContext context) {
+		SessionFactoryImplementor sessionFactoryImplementor = context.getSessionFactory();
+		ServiceRegistryImplementor registry = sessionFactoryImplementor.getServiceRegistry();
+
+		RemoteNeo4jDatastoreProvider provider = (RemoteNeo4jDatastoreProvider) registry.getService( DatastoreProvider.class );
+		createSequences( context.getDatabase(), context.getAllIdSourceKeyMetadata(), provider );
+		createEntityConstraints( provider.getDatabase(), context.getDatabase(), sessionFactoryImplementor.getProperties() );
+	}
+
+	private void createSequences(Database database, Iterable<IdSourceKeyMetadata> idSourceKeyMetadata, RemoteNeo4jDatastoreProvider provider) {
+		List<Sequence> sequences = sequences( database );
+
+		Statements constraintStatements = new Statements();
+		provider.getSequenceGenerator().createSequencesConstraints( constraintStatements, sequences );
+		provider.getSequenceGenerator().createUniqueConstraintsForTableSequences( constraintStatements, idSourceKeyMetadata );
+
+		StatementsResponse response = provider.getDatabase().executeQueriesInNewTransaction( constraintStatements );
+		List<StatementResult> results = response.getResults();
+		// TODO: Check response for errors
+
+		// We create the sequencese in a separate transaction because
+		// Neo4j does not allow the creation of constraints and graph elements in the same transaction
+		Statements sequenceStatements = new Statements();
+		provider.getSequenceGenerator().createSequences( sequenceStatements, sequences );
+		response = provider.getDatabase().executeQueriesInNewTransaction( sequenceStatements );
+		results = response.getResults();
+		// TODO: Check response for errors
```

Comment:
By the way, no need to add a coment for evey TODO (unless is something different), I will open a JIRA for them